src/v/storage/disk_log_impl.cc

```diff
@@ -1731,6 +1734,484 @@ disk_log_impl::make_cached_reader(log_reader_config config) {
+            } else {
+                // In this case we need to find the truncation point
+                // always starting from the beginning of the segment.
+                //
+                // prev is guaranteed to be smaller than target_size
+                // because we reached this branch.
+                auto prev = current_size - it->get()->file_size();
+                auto delta = target.target_size - prev;
+                truncate_after = delta;
+            }
+            auto last_index_entry = it->get()->index().find_above_size_bytes(
+              truncate_after);
+            if (
+              last_index_entry.has_value()
+              && model::prev_offset(last_index_entry->offset) > first) {
+                vlog(
+                  stlog.debug,
+                  "Setting offset range to {} - {}, {} bytes of the last "
+                  "segment are included",
+                  first,
+                  last_index_entry->offset,
+                  last_index_entry->filepos);
+
+                last_included_offset = model::prev_offset(
+                  last_index_entry->offset);
+
+                // We're including only part of the file so the total size
+                // of the range has to be adjusted
+                current_size
+                  -= (it->get()->file_size() - last_index_entry->filepos);
+            } else {
+                // There is no index entry that we can use to find the size
+                // of the offset range in this case
+                vlog(
+                  stlog.debug,
+                  "Setting offset range to {} - {} (end of segment), "
+                  "truncation point: {}, base_file_pos: {}",
+                  first,
+                  it->get()->offsets().committed_offset,
+                  truncate_after,
+                  first_segment_file_pos);
+                last_included_offset = it->get()->offsets().committed_offset;
+            }
+            break;
+        } else if (current_size > target.min_size) {
+            vlog(
+              stlog.debug,
+              "Setting offset range to {} - {}",
+              first,
+              it->get()->offsets().committed_offset);
+            // We can include full segment to the list of segments
+            last_included_offset = it->get()->offsets().committed_offset;
+            continue;
+        }
+    }
+
+    vlog(
+      stlog.debug,
+      "Discovered offset size: {}, last included offset: {}",
+      current_size,
+      last_included_offset);
+
+    if (current_size < target.min_size) {
+        vlog(stlog.debug, "Discovered offset range is not large enough");
+        last_included_offset = model::offset{};
+        current_size = 0;
+    }
+
+    if (segments.empty()) {
+        vlog(stlog.debug, "Can't find log segments to lock");
+        last_included_offset = model::offset{};
+        current_size = 0;
+    }
+
+    co_return offset_range_size_result_t{
```

Comment:
i believe this should either be an optional or lines like `current_size=0` be removed and let the caller decide if the result is too small and should be thrown away

let's have one responsible site for this. less code. less brain energy needed to understand who and why does things

Reply:
You still left `target.min_size`. I was suggesting that this could be removed completely and the above ```
    if (current_size < target.min_size) {
        vlog(stlog.debug, "Discovered offset range is not large enough");
        last_included_offset = model::offset{};
        current_size = 0;
    }
```

with it letting the caller do the if statement.

Reply:
changed return type to optional